### ISSUE TITLE
Only one device listed

### ISSUE THREAD

**Issue by grasgruenit** [OpenWonderLabs/SwitchBotAPI#323]
### Current Situation

When I use the API to list my devices, I only get my AC which is connected to a Hub MIni, but I have a Lock on an other Hub Mini and an infrared-device on a third Hub Mini. How can I list all devices?

### Logs

```markdown
"{"statusCode":100,"body":{"deviceList":[],"infraredRemoteList":[{"deviceId":"01-202401241540-43573798","deviceName":"Klimaanlage","remoteType":"Air Conditioner","hubDeviceId":"FE4D8B6A6FBE"}]},"message":"success"}"
```

### Configuration

```json
-
```

### Environment

Node-Red

**Comment by grasgruenit** [comment]
Anyone could give me a hint?

**Comment by grasgruenit** [comment]
Just a function node with this code:

`const crypto = global.get('crypto');
const https = global.get('https');

const token = "87bfb1c4b34a72fcb3ead99d156e13d133158723f213d155a6070d3e544e3bdf0afc5faf09b12dcf9a562d31a4d75e8e";
const secret = "0121b5e9ceaad020f3b428265cfad807";
const t = Date.now();
const nonce = "requestID";
const data = token + t + nonce;
const signTerm = crypto.createHmac('sha256', secret)
    .update(Buffer.from(data, 'utf-8'))
    .digest();
const sign = signTerm.toString("base64");
node.warn(sign);

const options = {
    hostname: 'api.switch-bot.com',
    port: 443,
    path: `/v1.1/devices`,
    method: 'GET',
    headers: {
        "Authorization": token,
        "sign": sign,
        "nonce": nonce,
        "t": t,
        'Content-Type': 'application/json'
    }
};

const req = https.request(options, function (res) {
    let responseData = '';
    res.on('data', function (chunk) {
        responseData += chunk;
    });
    res.on('end', function () {
        msg.payload = responseData;
        node.send(msg);
    });
});

req.on('error', function (error) {
    node.error(error);
});

req.end();
`

**Comment by donavanbecker** [comment]
Have you tried with other modules like Homebridge to see if you get the same results?

**Comment by grasgruenit** [comment]
No I have not as I have no Homekit / Homebridge running yet.

Probably will use an Zigbee IR Bridge for my project now.
https://www.zigbee2mqtt.io/devices/ZS06.html

**Comment by donavanbecker** [comment]
I would see if it is your function or the api by trying something else.

Have you tried this: https://github.com/tatuto017/node-red-contrib-switchbot-api

**Comment by grasgruenit** [comment]
No, I am not sure how to install it. In the NR palette, I can only install
the old 1.0 version.


/ Manuel Glojek | +436643256932
Zeit und Platz sparende Kurzsignatur. Ich bitte um Nachsicht für eventuell
fehlende persönliche Anrede und Grüße.


Am Di., 30. Juli 2024 um 17:59 Uhr schrieb Donavan Becker <
***@***.***>:
